crates/paimon/src/spec/schema_change.rs

```diff
@@ -0,0 +1,736 @@
+    }
+
+    /// impl the `add_column_with_comment_and_move`.
+    pub fn add_column_with_comment_and_move(
+        field_name: String,
+        data_type: DataTypeRoot,
+        comment: String,
+        move_: Move,
+    ) -> Self {
+        SchemaChange::AddColumn(AddColumn::new(
+            field_name,
+            data_type,
+            Some(comment),
+            Some(move_),
+        ))
+    }
+
+    /// impl the `rename_column`.
+    pub fn rename_column(field_name: String, new_name: String) -> Self {
+        SchemaChange::RenameColumn(RenameColumn::new(field_name, new_name))
+    }
+
+    /// impl the `drop_column`.
+    pub fn drop_column(field_name: String) -> Self {
+        SchemaChange::DropColumn(DropColumn::new(field_name))
+    }
+
+    /// impl the `update_column_type`.
+    pub fn update_column_type(field_name: String, new_data_type: DataTypeRoot) -> Self {
+        SchemaChange::UpdateColumnType(UpdateColumnType::new(field_name, new_data_type))
+    }
+
+    /// impl the `update_column_position`.
+    pub fn update_column_nullability(field_name: String, new_nullability: bool) -> Self {
+        SchemaChange::UpdateColumnNullability(UpdateColumnNullability {
+            field_names: vec![field_name],
+            new_nullability,
+        })
+    }
+
+    /// impl the `update_columns_nullability`.
+    pub fn update_columns_nullability(field_names: Vec<String>, new_nullability: bool) -> Self {
+        SchemaChange::UpdateColumnNullability(UpdateColumnNullability {
+            field_names,
+            new_nullability,
+        })
+    }
+
+    /// impl the `update_column_comment`.
+    pub fn update_column_comment(field_name: String, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment(UpdateColumnComment {
+            field_names: vec![field_name],
+            new_description: comment,
+        })
+    }
+
+    /// impl the `update_columns_comment`.
+    pub fn update_columns_comment(field_names: Vec<String>, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment(UpdateColumnComment {
+            field_names,
+            new_description: comment,
+        })
+    }
+
+    /// impl the `update_column_position`.
+    pub fn update_column_position(move_: Move) -> Self {
+        SchemaChange::UpdateColumnPosition(UpdateColumnPosition::new(move_))
+    }
+}
+
+/// The type of move.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L412>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub enum MoveType {
+    FIRST,
+    AFTER,
+    BEFORE,
+    LAST,
+}
+
+/// Represents a requested column move in a struct.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L410>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Move {
```

Comment:
Move is a keyword in rust, how about naming it `ColumnMove` and `ColumnMoveType`?

Reply:
> Move is a keyword in rust, how about naming it `ColumnMove` and `ColumnMoveType`?

OK, I will do it.